src/main/java/com/google/sps/data/UserQuery.java

```diff
@@ -103,37 +114,32 @@ public UserQuery(String query, String tags, String authors) {
    * present.
    */
   private Query.CompositeFilter createTagsFilter() {
-    // Implementation of collection interface methods to handle future change of ArrayList to set
-    Iterator<String> tagsAndIngredientsIterator = tagsAndIngredients.iterator();
-    
     // An array list of all the filters for tags/ingredients which can then be passed as a 
     // param to the composite filter constructor
     ArrayList<Query.Filter> tagsAndIngredientsFilters = new ArrayList<Query.Filter>();
     
     // Fills the arraylist with all the filters
-    while (tagsAndIngredientsIterator.hasNext()) {
+    for (String loopTag : tagsAndIngredients) {
```

Comment:
super nit: can you name this just tag? I know it's in a loop but that's not super relevant to what it actually represents.

ditto with loopAuthor below

Reply:
Ack and changed